x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1026 @@
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(false);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.onApiTokenUpdateSuccess({ ...newToken, type: ADMIN });
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        apiTokens: expect.any(Array),
+        activeApiToken: expect.any(Object),
+        activeApiTokenRawName: expect.any(String),
+        showCredentialsForm: expect.any(Boolean),
+      });
+    });
+  });
+
+  describe('setCredentialsData', () => {
+    const meta: IMeta = {
+      page: {
+        current: 1,
+        size: 1,
+        total_pages: 1,
+        total_results: 1,
+      },
+    };
+
+    describe('apiTokens', () => {
+      it('should set', () => {
+        mount();
+        CredentialsLogic.actions.setCredentialsData(meta, [newToken, newToken]);
+        expect(CredentialsLogic.values.apiTokens).toEqual([newToken, newToken]);
+      });
+    });
+
+    describe('meta', () => {
+      it('should be set', () => {
+        mount();
+        CredentialsLogic.actions.setCredentialsData(meta, [newToken, newToken]);
+        expect(CredentialsLogic.values.meta).toEqual(meta);
+      });
+    });
+
+    describe('isCredentialsDataComplete', () => {
+      it('should be set to true so we know that data fetching has completed', () => {
+        mount({
+          isCredentialsDataComplete: false,
+        });
+        CredentialsLogic.actions.setCredentialsData(meta, [newToken, newToken]);
+        expect(CredentialsLogic.values.isCredentialsDataComplete).toEqual(true);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setCredentialsData(meta, [newToken, newToken]);
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        apiTokens: expect.any(Array),
+        meta: expect.any(Object),
+        isCredentialsDataComplete: expect.any(Boolean),
+      });
+    });
+  });
+
+  describe('setCredentialsDetails', () => {
+    describe('isCredentialsDataComplete', () => {
+      it('should be set to true so that we know data fetching has been completed', () => {
+        mount({
+          isCredentialsDetailsComplete: false,
+        });
+        CredentialsLogic.actions.setCredentialsDetails(credentialsDetails);
+        expect(CredentialsLogic.values.isCredentialsDetailsComplete).toEqual(true);
+      });
+    });
+
+    describe('engines', () => {
+      it('should set `engines` from the provided details object', () => {
+        mount({
+          engines: [],
+        });
+        CredentialsLogic.actions.setCredentialsDetails(credentialsDetails);
+        expect(CredentialsLogic.values.engines).toEqual(credentialsDetails.engines);
+      });
+    });
+
+    describe('apiUrl', () => {
+      it('should set `engines` from the provided details object', () => {
```

Comment:
Typo
```suggestion
      it('should set `apiUrl` from the provided details object', () => {
```
Although ideally/eventually we shouldn't need apiUrl/this test anyway (per above comment)

Reply:
Removed apiUrl d1c2444c82b8cd0ac74afbdf583bc1c3a3654379